### PR TITLE
wallet-ext: sign message

### PR DESCRIPTION
**Pull request by pchrysochoidis**
## Description 

* wallet standard interface signMessage function
* reuse existing transaction requests for sign message requests
* started transitioning from transaction requests to approval requests

https://user-images.githubusercontent.com/10210143/223243789-d92965a1-3c1a-4499-a587-b22dff6a8080.mov

https://user-images.githubusercontent.com/10210143/223244077-38c17009-c2bd-4a9f-8142-1ab4e118de7e.mov

Button text updated Sign

Screenshot 2023-03-07 at 22 13 29

## Test Plan 

manual, use example app in wallet-adapters to sign a test message

closes APPS-510

### CODE REVIEW

**Review by mystie711 — APPROVED** [review]

**Comment by Jordan-Mysten** [comment]
Instead of `Approve` should we make the button `Sign`?

File: apps/wallet/src/shared/messaging/messages/payloads/transactions/ui/TransactionRequestResponse.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { type SuiSignMessageOutput } from '@mysten/wallet-standard/src';
```
Comment (Jordan-Mysten) [inline apps/wallet/src/shared/messaging/messages/payloads/transactions/ui/TransactionRequestResponse.ts:4]: Remove `/src`

File: apps/wallet/src/background/Transactions.ts [hunk 2]
```diff
@@ -5,26 +5,28 @@ import {
     type SignedTransaction,
     type SuiTransactionResponse,
 } from '@mysten/sui.js';
+import { type SuiSignMessageOutput } from '@mysten/wallet-standard/src';
```
Comment (Jordan-Mysten) [inline apps/wallet/src/background/Transactions.ts:8]: Remove `/src`

File: apps/wallet/src/shared/messaging/messages/payloads/transactions/ApprovalRequest.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    | {
+          type: 'v2';
+          justSign?: boolean;
+          //   TODO: Support transaciton builder string
+          //   data: SignableTransaction | string;
+          data: SignableTransaction;
+          options?: SuiSignAndExecuteTransactionOptions;
+          account: SuiAddress;
+      }
+    | { type: 'move-call'; data: MoveCallTransaction; account: SuiAddress }
```
Comment (Jordan-Mysten) [inline apps/wallet/src/shared/messaging/messages/payloads/transactions/ApprovalRequest.ts:30]: While you're here, if you rebase, you can remove the `move-call` and `serialized-move-call`.

Comment (Jordan-Mysten) [reply apps/wallet/src/shared/messaging/messages/payloads/transactions/ApprovalRequest.ts:30]: I forgot to do this earlier 😅

File: apps/wallet/src/ui/app/pages/approval-request/SignMessageRequest.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import { Text } from '../../shared/text';
+import { type SignMessageApprovalRequest } from '_payloads/transactions/ApprovalRequest';
+
+export type SignMessageRequestProps = {
+    request: SignMessageApprovalRequest;
+};
+
+export function SignMessageRequest({ request }: SignMessageRequestProps) {
+    const message = useMemo(() => {
+        return new TextDecoder().decode(fromB64(request.tx.message));
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/approval-request/SignMessageRequest.tsx:20]: I imagine there's a non-zero chance this causes useless output (in the event raw bytes are signed), might want a way to see the raw bytes being signed.

**Review by Jordan-Mysten — APPROVED** [review]
Some small feedback but LGTM overall

File: apps/wallet/src/shared/messaging/messages/payloads/transactions/ApprovalRequest.ts (hunk 3 above)
Comment (pchrysochoidis) [reply apps/wallet/src/shared/messaging/messages/payloads/transactions/ApprovalRequest.ts:30]: This requires many changes in the end will do it in another PR to be reviewed properly

Comment (pchrysochoidis) [reply apps/wallet/src/shared/messaging/messages/payloads/transactions/ApprovalRequest.ts:30]: https://mysten.atlassian.net/browse/APPS-611

File: apps/wallet/src/ui/app/pages/approval-request/SignMessageRequest.tsx (hunk 4 above)
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/pages/approval-request/SignMessageRequest.tsx:20]: https://mysten.atlassian.net/browse/APPS-612

**Pull request merged**